int32 type
int32 move_table
int32 move
---
int32 retu
